grpcutil/jwt_auth.go

```diff
@@ -39,22 +39,36 @@ type JWTAuthenticationHandler struct {
 	// the jwt ourselves. This flag skips the validation, and instead trusts the JWT has been validated
 	// upstream.
 	ValidationDisabled bool
+	Endpoint           string
 }
 
 // NewJWTAuthenticationHandler returns a JWTAuthenticationHandler with a verifier initialized. Uses defaults
 // for JWT related fields that will retreive a user email when using IAM on GCP.
-func NewJWTAuthenticationHandler(ctx context.Context, endpoint string, disableValidation bool) (*JWTAuthenticationHandler, error) {
-	verifier, err := jwtutil.NewVerifier(ctx, endpoint)
-	if err != nil {
-		return nil, err
-	}
-	return &JWTAuthenticationHandler{
-		Verifier:           verifier,
+func NewJWTAuthenticationHandler(ctx context.Context, opts ...HandlerOption) (*JWTAuthenticationHandler, error) {
+	j := &JWTAuthenticationHandler{
 		JWTPrefix:          "bearer ",
 		JWTKey:             "authorization",
 		PrincipalClaimKey:  "email",
-		ValidationDisabled: disableValidation,
-	}, nil
+		ValidationDisabled: false,
+	}
+	for _, opt := range opts {
+		opt(j)
```

Comment:
This will panic if opt is `nil` (which is possible since nil will satisfy that interface)